pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/codestyle/AbstractRegexNamingConventionsRule.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.rule.codestyle;
+
+import static net.sourceforge.pmd.properties.PropertyFactory.regexProperty;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import net.sourceforge.pmd.lang.apex.ast.ApexNode;
+import net.sourceforge.pmd.lang.apex.rule.AbstractApexRule;
+import net.sourceforge.pmd.properties.PropertyBuilder;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+
+public abstract class AbstractRegexNamingConventionsRule extends AbstractApexRule {
+    protected static final String CAMEL_CASE = "[a-z][a-zA-Z0-9]*";
```

Comment:
Those could directly be the compiled Pattern. It wouldn't do much, but you can't really do anything useful with those strings unless you compile them anyway. That's what the `defaultValue(String)` overload on the regex property builders does right away. Maybe also we could share them with Java in our StringUtil, in another PR

Reply:
Fixed